scummvm/scummvm
# SCUMM: Work around lava flowing up in MONKEY1-FLOPPY-VGA (Trac#10854)

See [Trac #10854](https://bugs.scummvm.org/ticket/10854) for more context and savegames.

In the floppy VGA version of Monkey Island 1 (*not* the CD version, not the EGA floppy version), the lava in the cave under the Giant Monkey Head currently flows up instead of down, with ScummVM. **This problem doesn't appear with the original interpreter.**

This effect uses color-cycling in the VGA version (where it's enabled… the Amiga version just disabled the effect, for example).

`initCycl()` does a different color-cycling for the older `GF_SMALL_HEADER` games. I wonder if `cycl->flags = 2;` shouldn't just always be `cycl->flags = 0;` for these games instead. But I can't prove that. It looks like it's been this way in ScummVM since commit cec02390ed9b068bb12c3ab03b0d92c7c70d8c4f (which was related to [Trac#1016](https://bugs.scummvm.org/ticket/1016)).

Diasm of the original floppy VGA interpreter could probably confirm or disprove this, but that's too much for me ;) The 2003 commit above says that LoomCD is another `GF_SMALL_HEADER ` game which also used this effect, but not in a way which makes sense (I think I can trigger one in Loom CD by doing `room 1` but epilepsy warning for this one).

So I suggest to just force the color-cycling effect to go in the opposite direction when we're in the MONKEY1-FLOPPY-VGA rooms where we know it's currently wrong.

## How to test

1. Start the floppy VGA version of Monkey Island 1 (which was only available for DOS, I believe)
2. `room 65` in the debugger, go to the right and look at the lava flowing from the wall. It should flow down, not up.
3. This effect also appears in `room 39` but you may need to try several tries since this room is randomy generated.

Another object using color-cycling in this game is Meathook's sign in room 48. But it looks like it just evenly alternates so I can't see any difference when changing the flags there.
